Cookbook/Proposals/Comments.md

```diff
@@ -0,0 +1,222 @@
+
+> Avoid block comments inline with code, as the code should be as self-documenting as possible. Exception: This does not apply to those comments used to generate documentation.
+
+> Avoid the use of C-style comments (/* ... */). Prefer the use of double- or triple-slash.
+
+
+## Motivation
+
+I think we sometimes overuse comments. The reason for this our StyleGuide doesn't say much on comments and this leads us to have different opinions whether a comment is worth to be added or not in our Pull Requests.
+
+I think comments are rather bad than good because of the 2 following reasons: 
+
+1. Comments eventually start to lie since the code frequently gets updated but comments remain unchanged
+2. Usage of comments gives a false-positive feeling to the author that the code is readable.
+
+Look at this comment:
+```swift
+/// consultation method, will be either 'phone' or 'video'
+public let consultationMethod: ConsultationMethod
+```
+
+Few lines below there's the enum declaration:
+```swift
+public enum ConsultationMethod: String {
+    case phone
+    case video
+    case faceToFace = "face to face"
+}
+```
+At the beginning, the `ConsultationMethod` had indeed 2 cases. This may be trivial here without big consequences however that not always a case and a lie may cost us many of hours.
+
+"Cross-files" comments are even more likely of becoming lies at any given point. Consider the following comment:
+```
+/// Make the booking confirmation screen. The created screen would always refetch
+/// the appointment from the backend.
+public func makeBookingConfirmation(
+    with appointmentId: AppointmentDTO.ID,
+    presentingOn flow: Flow,
+    dismiss: @escaping () -> Void
+) -> UIViewController 
+```
+Such a comment is a leak of information from another file into Builder. Builder doesn't know how BusinessController works. A change in a BusinessController can change the behavior of given screen, while there won't be a need to even enter the builder to find out there is a comment. In the end, comment stays there, unchanged.
+
+In our Pull Requests I've seen sometimes unnecessary comments
+```swift
+extension AppointmentDTO {
+    public enum State: String, Codable {
+        /// waiting for payment
+        case pending
+        /// user failed to pay for appoitment withing 10 min after the booking
+        case timedOut = "timed_out"
+        /// canceled by user or doctor
+        case cancelled
+        /// paid by user
+        case paid
+        /// doctor finished adding notes on the portal
+        case completed
+        /// patient did not answer the doctor's call
+        case noShow = "no_show"
+    }
+}
+```
+Can be simplified without any loss in understanding what the code represents to:
+
+```swift
+extension AppointmentDTO {
+    public enum State: String, Codable {
+        case waitingForPayment = "pending"
+        /// timeout == 10 min
+        case paymentTimedOut = "timed_out"
```

Comment:
Actually I find the original comment better than this open, since it provides context as to the condition that leads to the timeout (not just the time).

Reply:
Maybe it's a good exception for the comment. It's hard to say for me now as I didn't know what's important when I took this snippet on the same level the author knows while writing the code. 

However if that's a valid information from app perspective we can change the case naming to `case userFailedToPayInRequiredTime // timeOut == 10 min` to express that.